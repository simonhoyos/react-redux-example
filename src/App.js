import React from 'react';
import { connect } from 'react-redux';
import { INCREASE, DECREASE } from './reducers/countReducer';
import Posts from './components/Posts';
import './App.css';

function App({ count, increase, decrease }) {
  return (
    <div className="App">
      <button onClick={increase}>Increase</button>
      <span>{count}</span>
      <button onClick={decrease}>Decrease</button>
      <Posts />
    </div>
  );
}

const mapStateToProps = ({ countReducer }) => {
  return {
    count: countReducer.count,
  }
};

const mapDispatchToProps = dispatch => ({
  increase: () => dispatch({ type: INCREASE }),
  decrease: () => dispatch({ type: DECREASE }),
});

// const Something = connect(mapStateToProps, mapDispatchToProps);
// const Component = Something(App);
// <Component />

export default connect(mapStateToProps, mapDispatchToProps)(App);


// HOC o HOF = High Order Comoponent o Function
// function connect(mapStateToProps = {}, mapDispatchToProps = {}) {
//   la funcionalidad que queria reutilizar

//   return function Something(Component) {
//     return (
//       <Component {...mapStateToProps} {...mapDispatchToProps} >
//         {children}
//       </Component>
//     )
//   }
// }
