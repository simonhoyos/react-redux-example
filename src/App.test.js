import React from 'react';
import App from './App';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';

const middlewares = [thunk]
const mockStore = configureStore(middlewares);

it('renders without crashing', () => {
  const increase = jest.fn();
  const decrease = jest.fn();
  const getPosts = jest.fn();
  const store = mockStore({
    increase,
    decrease,
    getPosts,
    countReducer: { count: 0 },
    postsReducer: { posts: []}
  });

  const wrapper = mount(
    <Provider store={store}>
      <App store={store} />
    </Provider>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
