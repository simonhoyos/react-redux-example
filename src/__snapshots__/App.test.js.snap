// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <App
      count={0}
      decrease={[Function]}
      increase={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <div
        className="App"
      >
        <button
          onClick={[Function]}
        >
          Decrease
        </button>
        <span>
          0
        </span>
        <button
          onClick={[Function]}
        >
          Increase
        </button>
        <Connect(Posts)>
          <Posts
            getPosts={[Function]}
            posts={Array []}
          >
            <ul
              className="posts"
            />
          </Posts>
        </Connect(Posts)>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
